blockchain/blocks/src/tipset.rs

```diff
@@ -204,6 +205,32 @@ impl Tipset {
     }
 }
 
+impl Cbor for Tipset {
+    fn marshal_cbor(&self) -> Result<Vec<u8>, EncodingError> {
+        Ok(to_vec(&self.cids())?)
+    }
+}
+
+impl ser::Serialize for Tipset {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        (&self.blocks, &self.key).serialize(serializer)
+    }
+}
+
+impl<'de> de::Deserialize<'de> for Tipset {
+    fn deserialize<D>(deserializer: D) -> Result<Self, <D as Deserializer<'de>>::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let (blocks, key) = Deserialize::deserialize(deserializer)?;
+
+        Ok(Tipset { blocks, key })
+    }
+}
```

Comment:
This will create two different impls, the only reason marshal_cbor allows for overriding is for caching reasons, having two different impls will cause an issue later for sure